program/src/state/stake.rs

```diff
@@ -13,17 +15,14 @@ pub struct Stake {
     /// The discriminator is equal to `ArrayDiscriminator:: UNINITIALIZED` when
     /// the account is empty, and equal to `Stake::DISCRIMINATOR` when the account
     /// is initialized.
-    discriminator: [u8; 8],
+    pub discriminator: [u8; 8],
```

Comment:
This type is not being used on the client, so it is private in that sense.

Reply:
I'll leave it up to you, but generally for a type that should always have a specific implemented constant value for a certain field, its better to guard this behind an initializer method.

Reply:
What do you think about having a "manual" implementation of `Default` instead of the `new(..)` method:
```rust
impl Default for Stake {
    fn default() -> Self {
        Self {
            discriminator: Stake::SPL_DISCRIMINATOR.into(),
            amount: u64::default(),
            deactivation_timestamp: Option::default(),
            deactivating_amount: u64::default(),
            inactive_amount: u64::default(),
            authority: Pubkey::default(),
            validator: Pubkey::default(),
            last_seen_holder_rewards: u64::default(),
            last_seen_stake_rewards: u64::default(),
        }
    }
}
```

Reply:
Hmmm, never mind, it does not work.

Reply:
Ahh nuts. I think `new` is okay, where any of the fields you plan to initialize with, say `0`, are just that, and it takes the minimum necessary initialization values.